### PR TITLE
[ADLS] Account APIs object refactor and couple of bugfixes

### PR DESCRIPTION
**Pull request by ro-joowan**
### Swagger PR Links
* https://github.com/Azure/azure-rest-api-specs/pull/2417
* https://github.com/Azure/azure-rest-api-specs/pull/2391
* https://github.com/Azure/azure-rest-api-specs/pull/2263
* https://github.com/Azure/azure-rest-api-specs/pull/2353

### Changes in 2.4.0-preview

**Breaking changes**

- The `Account` operations object has been changed from `Account` to `Accounts`
    - E.g., `Account.Get(...)` to `Accounts.Get(...)`
- When creating or updating resources (`accounts`, `firewall rules`, etc.), explicit parameter objects are now required:
    - Account creation:
        - `DataLakeStoreAccount` to `CreateDataLakeStoreAccountParameters`
            - List of `FirewallRule` to `CreateFirewallRuleWithAccountParameters`
            - List of `TrustedIdProvider` to `CreateTrustedIdProviderWithAccountParameters`
    - Account update:
        - `DataLakeStoreUpdateParameters` to `UpdateDataLakeStoreParameters`
            - List of `FirewallRule` to `UpdateFirewallRuleWithAccountParameters`
            - List of `TrustedIdProvider` to `UpdateTrustedIdProviderWithAccountParameters`
    - Firewall rule creation and update:
        - `FirewallRule` to `CreateOrUpdateFirewallRuleParameters`
        - `FirewallRule` to `UpdateFirewallRuleParameters`
    - Trusted identity provider creation and update:
        - `TrustedIdProvider` to `CreateOrUpdateTrustedIdProviderParameters`
        - `TrustedIdProvider` to `UpdateTrustedIdProviderParameters`
- Bugfix: Removed the `childrenNum` field from `FileStatusProperties` because it was not supported
- Bugfix: Changed the data type of the `permission` field of `AclStatus` from `int` to `string`

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/_metadata/datalake-store_resource-manager.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-2018-01-04 20:02:08 UTC
+2018-02-05 18:29:47 UTC
 
 1) azure-rest-api-specs repository information
-GitHub user: Azure
-Branch:      current
-Commit:      db22f73d75fba57c9f1dcbd118bd1337ade842cf
+GitHub user: ro-joowan
+Branch:      revert-path-parameters
```
Comment (shahabhijeet) [inline src/SDKs/_metadata/datalake-store_resource-manager.txt]: this has to be generated from master branch

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/_metadata/datalake-store_resource-manager.txt (hunk 1 above)
Comment (ro-joowan) [reply src/SDKs/_metadata/datalake-store_resource-manager.txt]: @shahabhijeet Done.  I also updated the metadata files for ADLA -- the PR that you already merged: https://github.com/Azure/azure-sdk-for-net/pull/4043

**Comment by ro-joowan** [comment]
@shahabhijeet The last commit was to resolve merge conflicts

**Comment by ro-joowan** [comment]
@shahabhijeet Ping -- I don't think the build error is coming from my changes

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**